utilities/dataflow_hcapi_to_gcs_pubsub/dataflow_hcapi_to_gcs_pubsub.py

```diff
@@ -0,0 +1,380 @@
+            responseJSON = response.json()
+        except Exception as error:
+            reqSession.close()
+            responseFlag = "fail"
+            logging.error("process to get FHIR resource failed")
+            errorMessage = "Process to fetch FHIR resource from HCAPI failed due to : {}".format(str(type(error).__name__)+" --> "+ str(error))
+            responseJSON = dict()
+            responseJSON['errorMessage'] = errorMessage
+            return responseJSON,responseFlag
+        else:
+            reqSession.close()
+            return responseJSON,responseFlag
+
+    def hcapi_post(self, url:str, data:str):
+        """ This function to POST FHIR resource from FHIR store specified """
+        """
+        :param str url: HCAPI FHIR URL to fetch data
+        :param str data: JSON payload string
+        :return dict responseJSON: response in JSON based on criteria
+        :return str responseFlag: repsonse status pass or fail
+        """
+        try:
+            responseFlag = "success"
+            logging.info("creating headers with new token for HCAPI POST request")
+            creds, project = google.auth.default()
+            auth_req = google.auth.transport.requests.Request()
+            creds.refresh(auth_req)
+            authToken = creds.token
+            authHeaders = {
+                "Authorization": f"Bearer {authToken}",
+                "Prefer": "handling=strict"
+            }
+            logging.info("Posting FHIR resource to HCAPI")
+            payload_data =json.loads(data)
+            reqSession = self.createRequestSession()
+            response = requests.post(url, headers=authHeaders, json=payload_data,timeout=30)
+            response.raise_for_status()
+            responseJSON = response.json()
+        except Exception as error:
+            reqSession.close()
+            logging.error("process to POST FHIR resource failed")
+            errorMessage = "Process to POST FHIR Resource to HCAPI failed due to : {}".format(str(type(error).__name__)+" --> "+ str(error))
+            responseJSON = dict()
+            responseJSON['errorMessage'] = errorMessage
+            responseFlag = "fail"
+            return responseJSON,responseFlag
+        else:
+            reqSession.close()
+            return responseJSON,responseFlag
+
+    def post_fhir_message(self, resourceType:str,payload:str):
+        """ Function to post messages to FHIR store """
+        """
+        :param str resourceType: FHIR Resource to post data
+        :param str payload: FHIR Resource as string
+        :return dict responseJSON: response in JSON based on criteria
+        :return str responseFlag: repsonse status pass or fail
+        """
+        resourceType=str(resourceType)
+        url = self.hcapi_fhir_url(store=self.hcapi_post_fhir_store)+f'/{resourceType}'
+        postResponse,responseFlag = self.hcapi_post(url,payload)
+        return postResponse,responseFlag
```

Comment:
do we need post functions here?

Reply:
As per discussion added comment for future reference.